extensions-contrib/kafka-emitter/src/main/java/org/apache/druid/emitter/kafka/KafkaEmitter.java

```diff
@@ -130,24 +120,24 @@ public void start()
 
   private void sendMetricToKafka()
   {
-    sendToKafka(config.getMetricTopic(), metricQueue);
+    sendToKafka(config.getMetricTopic(), metricQueue, setProducerCallback(metricLost));
   }
 
   private void sendAlertToKafka()
   {
-    sendToKafka(config.getAlertTopic(), alertQueue);
+    sendToKafka(config.getAlertTopic(), alertQueue, setProducerCallback(alertLost));
   }
 
-  private void sendToKafka(final String topic, MemoryBoundLinkedBlockingQueue<String> recordQueue)
+  private void sendToKafka(final String topic, MemoryBoundLinkedBlockingQueue<String> recordQueue, Callback callback)
   {
     ObjectContainer<String> objectToSend;
     try {
       while (true) {
         objectToSend = recordQueue.take();
-        producer.send(new ProducerRecord<>(topic, objectToSend.getData()), producerCallback);
+        producer.send(new ProducerRecord<>(topic, objectToSend.getData()), callback);
       }
     }
-    catch (InterruptedException e) {
+    catch (Throwable e) {
       log.warn(e, "Failed to take record from queue!");
```

Comment:
on any  exception  from the block inside `while(true) { ... }` would lead to event emission getting stopped. Does it make sense to have a try-catch inside there to ignore exceptions except for `InterruptedException` so that emitting stops only on thread interrupt which would happen when druid is shutting down.

Reply:
@himanshug, thanks for the review. As the potential exceptions from the producer are handled in the `callback`, so in my opinion, currently the `while(true) { ... }` block is already supposed to raised the `InterruptedException` only.

Reply:
sounds reasonable, however I would still make slight log messaage change just in case.

Reply:
thanks @himanshug, it's more clear now.